lib/Dialect/TTKernel/IR/TTKernelOps.cpp

```diff
@@ -91,6 +91,78 @@ ::mlir::LogicalResult CBWaitFrontOp::verify() {
   return success();
 }
 
+::mlir::LogicalResult TilizeInitOp::verify() {
+  if (not insideDispatchOpRegion(getOperation())) {
+    return emitOpError("TilizeInitOp must be inside of a DispatchOp region");
+  }
+  if (getCbIn().getType().getPort() == getCbOut().getType().getPort()) {
+    return emitOpError("cbIn and cbOut must be different");
+  }
+  if (mlir::isa<tt::TileType>(
+          getCbIn().getType().getMemref().getElementType())) {
+    return emitOpError("cbIn must have scalar element type");
+  }
+  if (not mlir::isa<tt::TileType>(
+          getCbOut().getType().getMemref().getElementType())) {
+    return emitOpError("cbOut must have tile element type");
+  }
+  return success();
+}
+
+::mlir::LogicalResult UntilizeInitOp::verify() {
+  if (not insideDispatchOpRegion(getOperation())) {
+    return emitOpError("TilizeInitOp must be inside of a DispatchOp region");
+  }
+  if (getCbIn().getType().getPort() == getCbOut().getType().getPort()) {
+    return emitOpError("cbIn and cbOut must be different");
+  }
+  if (not mlir::isa<tt::TileType>(
+          getCbIn().getType().getMemref().getElementType())) {
+    return emitOpError("cbIn must have tile element type");
+  }
+  if (mlir::isa<tt::TileType>(
+          getCbOut().getType().getMemref().getElementType())) {
+    return emitOpError("cbOut must have scalar element type");
+  }
+  return success();
+}
+
+::mlir::LogicalResult TilizeBlockOp::verify() {
+  if (not insideDispatchOpRegion(getOperation())) {
+    return emitOpError("TilizeBlockOp must be inside of a DispatchOp region");
+  }
+  if (getCbIn().getType().getPort() == getCbOut().getType().getPort()) {
+    return emitOpError("cbIn and cbOut must be different");
+  }
+  if (mlir::isa<tt::TileType>(
+          getCbIn().getType().getMemref().getElementType())) {
+    return emitOpError("cbIn must have scalar element type");
+  }
+  if (not mlir::isa<tt::TileType>(
+          getCbOut().getType().getMemref().getElementType())) {
+    return emitOpError("cbOut must have tile element type");
+  }
+  return success();
+}
+
+::mlir::LogicalResult UntilizeBlockOp::verify() {
+  if (not insideDispatchOpRegion(getOperation())) {
+    return emitOpError("TilizeBlockOp must be inside of a DispatchOp region");
+  }
+  if (getCbIn().getType().getPort() == getCbOut().getType().getPort()) {
+    return emitOpError("cbIn and cbOut must be different");
+  }
+  if (not mlir::isa<tt::TileType>(
+          getCbIn().getType().getMemref().getElementType())) {
+    return emitOpError("cbIn must have tile element type");
+  }
+  if (mlir::isa<tt::TileType>(
+          getCbOut().getType().getMemref().getElementType())) {
+    return emitOpError("cbOut must have scalar element type");
+  }
+  return success();
```

Comment:
This whole block repeats for UntilizeBlockOp and UntilizeInitOp. Let's extract it to a function. Same can be done for TilizeInitOp & TilizeBlockop.